service/share/set.go

```diff
@@ -0,0 +1,69 @@
+package share
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// limitedSet is a thread safe set of peers with given limit.
+// Inspired by libp2p peer.Set but extended with Remove method.
+type limitedSet struct {
+	lk sync.RWMutex
+	ps map[peer.ID]struct{}
+
+	limit int
```

Comment:
Will this ever be negative?

Reply:
https://github.com/celestiaorg/celestia-node/pull/799/commits/814b3b677daf7cbffa5b987f1a97611aeeb1cfa3